src/platform/silabs/wifi/icd/PlatformSleepManager.cpp

```diff
@@ -0,0 +1,126 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app/icd/server/ICDConfigurationData.h>
+#include <lib/support/logging/CHIPLogging.h>
+#include <platform/silabs/wifi/WifiInterfaceAbstraction.h>
+#include <platform/silabs/wifi/icd/PlatformSleepManager.h>
+
+using namespace chip::app;
+
+namespace chip {
+namespace DeviceLayer {
+namespace Silabs {
+
+// Initialize the static instance
+PlatformSleepManager PlatformSleepManager::mInstance;
+
+CHIP_ERROR PlatformSleepManager::Init()
+{
+    return CHIP_NO_ERROR;
+}
+
+void PlatformSleepManager::HandleCommissioningComplete()
+{
+    TransitionToLowPowerMode();
+}
+
+void PlatformSleepManager::HandleInternetConnectivityChange()
+{
+    if (!isCommissioningInProgress)
```

Comment:
I do not have a clear state machine in mind, if we loose internet during commissioning, the path is that HandleCommissioningSessionStopped will be called and eventually and thus bring the chip back to sleep?

Reply:
This isn't a case that is currently supported by the existing wi-fi implementation so that would be an edge case that isn't covered.

The goal of the PR is just to move the existing logic to into a centralized spot.